### PR TITLE
Implement the repo.Latest method of the new charm store.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

File: charmrepo/charmstore.go [hunk 1]
```diff
@@ -71,7 +74,58 @@ func (s *CharmStore) Get(curl *charm.URL) (charm.Charm, error) {
...
+
+	// Prepare the request to the charm store.
+	urls := make([]string, len(curls))
+	ids := make([]string, len(curls))
+	for i, curl := range curls {
+		url := curl.WithRevision(-1).String()
+		urls[i] = url
+		ids[i] = "id=" + url
+	}
+	path := "/meta/any?include=id-revision&include=hash256&" + strings.Join(ids, "&")
```
Comment (mhilton) [inline charmrepo/charmstore.go]: Perhaps you could build this with a URL e.g.
```
var u url.URL
u.Path = "/meta/any"
var v url.Values
v.Add("include", "id-revision")
v.Add("include", "hash256")
for i, curl := range curls {
        url := curl.WithRevision(-1).String()
        urls[i] = url
        v.Add("id", url)
}
u.RawQuery = v.String()
path = u.String()
```
?

**Comment by mhilton** [comment]
:+1: LGTM with a suggestion.

File: charmrepo/charmstore.go [hunk 2]
```diff
@@ -71,7 +74,58 @@ func (s *CharmStore) Get(curl *charm.URL) (charm.Charm, error) {
...
+
+	// Execute the request and retrieve results.
+	var results map[string]struct {
```
Comment (rogpeppe) [inline charmrepo/charmstore.go]: I think you could avoid some work here by declaring
the retrieved fields explicitly:

    var results map[string] struct {
      Meta struct {
           IdRevision params.IdRevisionResponse `json:"id-revision"`
           Hash256 params.HashResponse `json:"hash256"`
      }
    }

Even better would be to add a bulk meta method as a companion
to csclient.Client.Meta.

File: charmrepo/charmstore.go (hunk 1 above)
Comment (rogpeppe) [inline charmrepo/charmstore.go]: I had been thinking something similar.

**Comment by rogpeppe** [comment]
LGTM with one suggestion, thanks!

File: charmrepo/charmstore.go (hunk 2 above)
Comment (frankban) [inline charmrepo/charmstore.go]: Nice suggestion! Done.
csclient.BulkMeta is also a good idea, I'll investigate that later.

File: charmrepo/charmstore.go (hunk 1 above)
Comment (frankban) [inline charmrepo/charmstore.go]: Nice, Done.

**Comment by frankban** [comment]
Thanks for the reviews!

**Pull request merged**